doc/rtd/howto/debugging.rst

```diff
@@ -0,0 +1,179 @@
+.. _how_to_debug:
+
+How to debug cloud-init
+***********************
+
+There are several cloud-init :ref:`failure modes<failure_states>` that one may
+need to debug. Debugging is specific to the scenario, but the starting points
+are often similar:
+
+* :ref:`I cannot log in<cannot_log_in>`
+* :ref:`Cloud-init did not run<did_not_run>`
+* :ref:`Cloud-init did the unexpected<did_not_do_the_thing>`
+* :ref:`Cloud-init never finished running<did_not_finish_running>`
+
+.. _cannot_log_in:
+
+I can't log in to my instance
+=============================
+
+One of the more challenging scenarios to debug is when you don't have
+shell access to your instance. You have a few options:
+
+1. Acquire log messages from the serial console and check for any errors.
+
+2. To access to instances without SSH access, create a user with password
+   access (using the user-data) and log in via the cloud serial port console.
+   This only works if ``cc_users_groups`` successfully ran.
+
+3. Try running the same user-data locally, such as in one of the
+   :ref:`tutorials<tutorial_index>`. Use LXD or QEMU locally to get a shell or
+   logs then debug with :ref:`these steps<did_not_do_the_thing>`.
+
+4. Try copying the image to your local system, mount the filesystem locally
+   and inspect (create a chroot in the image filesystem before attempting the
+   steps in other sections).
+
+.. _did_not_run:
+
+Cloud-init did not run
+======================
+
+1. Check the output of ``cloud-init status --long``
+
+   - what is the value of the ``'extended_status'`` key?
+   - what is the value of the ``'boot_status_code'`` key?
+
+See :ref:`here<reported_status>` for more information on the status.
```

Comment:
```suggestion
See :ref:`our reported status explanation<reported_status>` for more information.
```

Reply:
(I'm suggesting changing all these because using "here" as linking text is not considered best accessibility practice)